File: fishy.java
import java.util.Scanner;

public class fishy {

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		
		// Read in the number of days to process.
		int n = in.nextInt();
		
		// Loop through each day.
		for(int curDay = 1; curDay <= n; ++curDay) {
			// Read the number of fish.
			int num = in.nextInt();
			
			// Get the first power of two greater than or equal to the number.
			int powerOf2 = (int)(Math.ceil(Math.log(num)/Math.log(2)));
			int fishTree =  1 << powerOf2;
			
			// Find the number of fish remaining on the tree.
			int remaining = fishTree - num;
			
			// Print the answer.
			System.out.printf("Day #%d: %d extra fish\n", curDay, remaining);
		}
		
		in.close();
	}
}

File: fishy.c
#include <math.h>
#include <stdio.h>

int main() {
    
    // Read in the number of days to process.
    int numCases;
    scanf("%d", &numCases);
    
    // Loop through each day.
    for(int curCase = 1; curCase <= numCases; ++curCase) {
        // Read the number of fish.
        int numFish;
        scanf("%d", &numFish);
        
        // Get the first power of two greater than or equal to the number.
        int fishTree = 1;
        while(fishTree < numFish)
            fishTree *= 2;
        
        // Find the number of fish remaining on the tree.
        int remaining = fishTree - numFish;
        
        // Print the answer.
        printf("Day #%d: %d extra fish\n", curCase, remaining);
    }
}
